python/ray/tune/tests/test_trial_scheduler.py

```diff
@@ -1080,9 +1091,8 @@ def testPerturbsLowPerformingTrialsSynch(self):
     def testPerturbWithoutResample(self):
         pbt, runner = self.basicSetup(resample_prob=0.0)
         trials = runner.get_trials()
-        self.assertEqual(
-            pbt.on_trial_result(runner, trials[0], result(20, -100)),
-            TrialScheduler.CONTINUE)
+        self.on_trial_result(pbt, runner, trials[0], result(20, -100),
+                             TrialScheduler.PAUSE)
```

Comment:
Previously we expected CONTINUE here, why are we expected PAUSE now?

Reply:
Thanks for looking so closely!
I think ideally we want to update this test to not be testing code line by line but rather focusing more on the scheduler e2e behavior. This is hard without a clear API on scheduler and runner, which we are de-prioritizing for now. 
Going back to these two tests. So we have 5 trials running by the time `basicSetup` is done. Invoking `on_trial_result` on `trials[0]` with a bad result would cause it to be replaced by better performing trials. Before, this is done through `reset` + `restore` path; now because of this (see PR description #2), every trial is stopped and then started. So trials[0] leaves pbt scheduler with PAUSED state.